model-optimizer/extensions/back/InterpolateReshape.py

```diff
@@ -0,0 +1,150 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import numpy as np
+
+from extensions.ops.elementwise import Mul
+from extensions.ops.gather import Gather
+from mo.back.replacement import BackReplacementPattern
+from mo.front.caffe.extractors.utils import get_canonical_axis_index
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs, create_op_node_with_second_input
+from mo.graph.graph import Graph
+from mo.ops.shape import Shape
+
+
+class InterpolateConcat(BackReplacementPattern):
+    """
+    Replaces hard-coded 1-port input of Interpolate with reshape-able sub-graph using the following Concat inputs
+
+    BEFORE:
+            input                   Const
+    shape=[1, 3, 30, 40]      value=[60, 160]
+            \                   /
+           Interpolate(axes=(2, 3))     input_1
+            shape=[1, 3, 60, 160]    shape=[1, 4, 60, 160]
+                        \           /
+                        Concat(axis=1)
+                    shape=[1, 7, 60, 160]
+    AFTER:
+                input
+            shape=[1, 3, 30, 40]           input_1
+               |                     shape=[1, 4, 60, 160]
+               |                      /        |
+               |                  ShapeOf      |
+               |                    |          |
+               |               Gather          |
+               |     indices=(2, 3); axis=0    |
+               \                    |          |
+                Interpolate(axes=(2, 3))      |
+            shape=[1, 3, 60, 160]             |
+                        \                   /
+                           Concat(axis=1)
+                        shape=[1, 7, 60, 160]
+
+    """
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['cmd_params'].generate_experimental_IR_V10,
+                       lambda graph: graph.graph['cmd_params'].keep_shape_ops]
+    force_clean_up = True
+
+    @staticmethod
+    def make_interpolate_reshapeable(node):
+        concat = node.out_port(0).get_destination().node
```

Comment:
Put an assert here that node is of type 'Interpolate' (rename the variable to 'interpolate'?) and the `concat` is of type Concat.

nit: May be it make sense to add the second parameter to this function where the 'Concat' node is passed.